courier/message.go

```diff
@@ -16,14 +21,146 @@ const (
 	MessageStatusAbandoned
 )
 
+const (
+	messageStatusQueuedText     = "queued"
+	messageStatusSentText       = "sent"
+	messageStatusProcessingText = "processing"
+	messageStatusAbandonedText  = "abandoned"
+)
```

Comment:
```suggestion
// A Message's Status
//
// swagger:enum messageStatus
type messageStatus string

const (
	messageStatusQueuedText messageStatus     = "queued"
	messageStatusSentText messageStatus       = "sent"
	messageStatusProcessingText messageStatus = "processing"
	messageStatusAbandonedText messageStatus  = "abandoned"
)
```

You might be able to avoid the patches if you do it this way. What is important is that the type name is the same as the `// swagger:enum name`, and that all values are in one block immediately after the type for this to work. But you can also just keep the patch if this does not work properly.

Reply:
@zepatrik That would definitely make us avoid the patches, but is it really necessary to introduce another type just for this ? I think that might complicate code-base a little bit more (all the need for casting `messageStatus` to `string` and vice versa, new `IsValid()` method just for `messageStatus` etc.) ? We already have 150 lines of code just for two enumeration, i think expanding it might be a little bit confusing to future developers of this project. What do you think ?